source/PreferencesPanel.cpp

```diff
@@ -179,6 +205,31 @@ bool PreferencesPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &comma
 		if(zones[latest].Value().KeyName() != Command::MENU.KeyName())
 			Command::SetKey(zones[latest].Value(), 0);
 	}
+	else if(key == 'd')
+	{
+		page = 'i';
+		if(!downloadedInfo)
+		{
+			if(!Plugins::Download(PLUGIN_LIST_URL, Files::Config() + "plugins.json"))
+				printf("Failed to download Plugin-List from %s\n", PLUGIN_LIST_URL.c_str());
```

Comment:
This definitely needs to be a message to the player. I would suggest adding a text in the middle of the plugin panel or at the top with the message, ideally with a button to retry.

Reply:
ok